src/service.js

```diff
@@ -0,0 +1,316 @@
+if(!global._babelPolyfill) { require('babel-polyfill'); }
+
+import Proto from 'uberproto';
+import mongoose from 'mongoose';
+import filter from 'feathers-query-filters';
+import errors from 'feathers-errors';
+import errorHandler from './error-handler';
+
+// Use native promises
+mongoose.Promise = global.Promise;
+
+const Schema = mongoose.Schema;
```

Comment:
If we decide to force the user to pass in a mongoose model then we can remove mongoose as a production dependency. Another plus is that we can get rid of the connection logic and the logic around creating a mongoose model.

However, I'm unsure if we'll be able to support our current hook definition and we'll lose our nice clean object literal format of a model. Wanting some feedback here @marshallswain @corymsmith @daffl.